samples/minimum_criteria.rst

```diff
@@ -0,0 +1,82 @@
+.. _definition_and_minimum_criteria:
+
+Sample Definition and Minimum Criteria
+######################################
+
+Definition
+==========
+
+A sample is a basic Zephyr application that provides a brief overview of a feature, subsystem, or module. For example, Kernel services, drivers, protocols, etc.
+Samples are limited in scope and should focus only on capturing the nontrivial and introductory aspects of the functionality in question.
+
+Minimum Requirements/ Criteria for Samples:
+===========================================
+
+1. Samples must not use Zephyr's testing framework.
++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Do not use :kconfig:option:`CONFIG_ZTEST` or :kconfig:option:`CONFIG_ZTEST_NEW_API`.
+  * Should not use zasserts or asserts in samples.
+
+2. We shall not use the sample as a testcase.
++++++++++++++++++++++++++++++++++++++++++++++
+  * A feature should already have testcases under the tests category before the sample gets added. Testcases are mandatory, and having samples may be optional.
+  * The primary purpose of a sample is to provide a reference to the user.
+
+3. Should ideally run on multiple platforms and architectures.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Minimize use of platform_allow and architecture filters as it would limit the testing on the mentioned platforms and architectures. Reference: `Testcase Docs`_
```

Comment:
can you wrap the line to 100 chars per line?

Reply:
Done